packages/core/util/Job.js

```diff
@@ -206,6 +206,19 @@ class Job {
 			this.fn(...args);
 		}
 	}
+
+	promise = (promise) => {
```

Comment:
Don't want to start exploring `Observable` for this PR, but it might be worth using it for canceling.